dataRetriever/dataPool/headersCache/headersPool_test.go

```diff
@@ -0,0 +1,342 @@
+package headersCache_test
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever/dataPool/headersCache"
+	"github.com/stretchr/testify/assert"
+)
+
+func createASliceOfHeaders(numHeaders int, shardId uint32) ([]block.Header, [][]byte) {
+	headers := make([]block.Header, 0)
+	headersHashes := make([][]byte, 0)
+	for i := 0; i < numHeaders; i++ {
+		headers = append(headers, block.Header{Nonce: uint64(i), ShardId: shardId})
+		headersHashes = append(headersHashes, []byte(fmt.Sprintf("%d_%d", shardId, i)))
+	}
+
+	return headers, headersHashes
+}
+
+func createASliceOfHeadersNonce0(numHeaders int, shardId uint32) ([]block.Header, [][]byte) {
+	headers := make([]block.Header, 0)
+	headersHashes := make([][]byte, 0)
+	for i := 0; i < numHeaders; i++ {
+		headers = append(headers, block.Header{Nonce: 0, ShardId: shardId})
+		headersHashes = append(headersHashes, []byte(fmt.Sprintf("%d_%d", shardId, i)))
+	}
+
+	return headers, headersHashes
+}
+
+func TestNewHeadersCacher_AddHeadersInCache(t *testing.T) {
+	t.Parallel()
+
+	hdrsCacher, _ := headersCache.NewHeadersCacher(1000, 100)
+
+	nonce := uint64(1)
+	shardId := uint32(0)
+
+	hdrHash1 := []byte("hash1")
+	hdrHash2 := []byte("hash2")
+	testHdr1 := &block.Header{Nonce: nonce, ShardId: shardId}
+	testHdr2 := &block.Header{Nonce: nonce, ShardId: shardId, Round: 100}
+
+	hdrsCacher.Add(hdrHash1, testHdr1)
+	hdrsCacher.Add(hdrHash2, testHdr2)
+
+	hdr, err := hdrsCacher.GetHeaderByHash(hdrHash1)
+	assert.Nil(t, err)
```

Comment:
Since you use more `asserts` in a row, replace them with `require`. `testify/require`.

Reply:
done.